packages/itwin/tree-widget/src/components/trees/SpatialTree.tsx

```diff
@@ -2,27 +2,18 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-import type { SpatialContainmentTreeProps } from "@itwin/appui-react";
-import { SpatialContainmentTree } from "@itwin/appui-react";
-import { useResizeObserver } from "@itwin/core-react";
-import React, { useCallback, useState } from "react";
+import { SpatialContainmentTree, useActiveIModelConnection } from "@itwin/appui-react";
+import React from "react";
+import type { SpatialTreeProps } from "../../types";
+import { AutoSizer } from "../utils/AutoSizer";
 
 export const SpatialTreeComponent = (
-  props: Omit<SpatialContainmentTreeProps, "width" | "height">
-) => {
-  const [height, setHeight] = useState(0);
-  const [width, setWidth] = useState(0);
-  const handleResize = useCallback((w: number, h: number) => {
-    setHeight(h);
-    setWidth(w);
-  }, []);
-  const ref = useResizeObserver<HTMLDivElement>(handleResize);
+  props: SpatialTreeProps) => {
+  const iModel = useActiveIModelConnection();
 
-  return (
-    <div ref={ref} style={{ width: "100%", height: "100%" }}>
-      {width && height && (
-        <SpatialContainmentTree {...props} width={width} height={height} />
-      )}
-    </div>
+  return !iModel ? null : (
```

Comment:
seems more straightforward to do:

iModel && (<AutoSizer> ... </AutoSizer>)

Reply:
Agreed, but that causes issues with the SelectableContentDefintion.
![image](https://user-images.githubusercontent.com/11051042/158473494-dd1594f2-c2cb-4689-8f89-fd851c507aa9.png)
![image](https://user-images.githubusercontent.com/11051042/158473525-22960e3a-bdfb-478c-b906-e43d9afbf762.png)

Reply:
Eh actually nvm, i guess I can just wrap the entire thing in a react fragment so its a valid ReactNode; Ill update